### PR TITLE
IR Public SDK update

### PR DESCRIPTION
**Pull request by AshwaniBansal1**
IR SDK update

### CODE REVIEW

File: js/src/launchResponse.ts [hunk 1]
```diff
@@ -4,5 +4,8 @@
 export type LaunchResponse = {
     container: HTMLDivElement;
     sessionId: string;
+    readerReadyDuration: number;
+    launchDuration: number;
+    gcmCorrelationId: string;
     charactersProcessed: number;
```
Comment (cristobal-buenrostro) [inline js/src/launchResponse.ts:10]: please keep this order for the properties (exactly as internal sdk):

container
charactersProcessed
readerReadyDuration
launchDuration
gcmCorrelationId
sessionId

File: js/src/launchAsync.ts [hunk 2]
```diff
@@ -85,6 +90,7 @@ export function launchAsync(token: string, subdomain: string, content: Content,
 
         isLoading = true;
         const startTime = Date.now();
+        launchResponseError = null;
         options = {
```
Comment (cristobal-buenrostro) [inline js/src/launchAsync.ts:94]: please leave one empty line between 93 and 94

**Comment by cristobal-buenrostro** [comment]
Hi @AshwaniBansal1 

Two additional changes to consider:

- You missed changes for **/tests/launchAsync.tests.ts** file, please check and apply the changes from internal sdk PR.
- Update **/js/readme.md** file to mention the new version 1.3.0

File: js/src/error.ts [hunk 3]
```diff
@@ -4,6 +4,9 @@
 export type Error = {
     code: ErrorCode;
     message: string; // Human-readable representation of the error
+    readerReadyDuration?: number; // Elapsed time since launchAsync call until reader request succeeded
```
Comment (rwallerms) [inline js/src/error.ts:7]: It was not expected that these changes would go into Public SDK. They were Internal SDK only

Comment (rwallerms) [reply js/src/error.ts:7]: Main thing we needed in the Public SDK update were the microphone permissions update

**Review by rwallerms — CHANGES_REQUESTED** [review]

**Pull request closed without merging**